internal/pkg/pluginmanager/manager.go

```diff
@@ -25,37 +27,49 @@ func DownloadPlugins(conf *configloader.Config) error {
 
 	for _, tool := range conf.Tools {
 		pluginFileName := configloader.GetPluginFileName(&tool)
+		pluginMD5FileName := configloader.GetPluginMD5FileName(&tool)
+		// plugin does not exist
 		if _, err := os.Stat(filepath.Join(pluginDir, pluginFileName)); errors.Is(err, os.ErrNotExist) {
-			// plugin does not exist
-			err := dc.download(pluginDir, pluginFileName, tool.Plugin.Version)
-			if err != nil {
+			// download .so file
+			if err := dc.download(pluginDir, pluginFileName, tool.Plugin.Version); err != nil {
+				return err
+			}
+			// download .md5 file
+			if err := dc.download(pluginDir, pluginMD5FileName, tool.Plugin.Version); err != nil {
+				return err
+			}
+			// check if the downloaded plugin md5 matches with .md5
+			if err := checkPluginMismatch(pluginDir, pluginFileName, pluginMD5FileName, tool.Name); err != nil {
 				return err
 			}
 			continue
 		}
-		// check md5
-		dup, err := checkFileMD5(filepath.Join(pluginDir, pluginFileName), dc, pluginFileName, tool.Plugin.Version)
+
+		// if .so exists
+		isMD5Match, err := md5helper.ValidateFileMatchMD5(filepath.Join(pluginDir, pluginFileName), filepath.Join(pluginDir, pluginMD5FileName))
 		if err != nil {
 			return err
 		}
-
-		if dup {
+		// if .so exists, and matches with .md5, continue
+		if isMD5Match {
 			log.Infof("Plugin: %s already exists, no need to download.", pluginFileName)
 			continue
 		}
-
-		log.Infof("Plugin: %s changed and will be overrided.", pluginFileName)
-		if err = os.Remove(filepath.Join(pluginDir, pluginFileName)); err != nil {
+		// if .so exists, but mismatches with .md5, re-download
+		log.Infof("Plugin: %s doesn't match with dtm core and will be downloaded.", pluginFileName)
```

Comment:
The log message is not accurate. The plugin doesn't match its own MD5 sum, not with dtm core. Here it has nothing to do with the core, right?